[Xcode] Enable compilation caching by default

emw-apple opened the pull request:
#### 4df1e0ec0e7d4eb880d2187934d764af060cfb23
<pre>
[Xcode] Enable compilation caching by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=303579">https://bugs.webkit.org/show_bug.cgi?id=303579</a>
<a href="https://rdar.apple.com/131590077">rdar://131590077</a>

Reviewed by Alexey Proskuryakov.

Toggle COMPILATION_CACHE_ENABLE_CACHING in CommonBase.xcconfig. Add a
script phase that runs in WTF which ensures that WebKit&apos;s opt-in ccache
support is not simultaneously enabled -- the two caches are redundant
when used together.

* Configurations/CommonBase.xcconfig:
* Makefile.shared: Remove logic that would automatically set
  WK_USE_CCACHE.
* Source/WTF/Scripts/check-ccache-compilation-caching.sh: Added.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/mac/CocoaAccessibilityConstantsPrivate.h: Added.
* Tools/ccache/README.md: Leave a note about LLVM compilation caching
  being the new default.

Canonical link: <a href="https://commits.webkit.org/304172@main">https://commits.webkit.org/304172@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/11ca2bf3d82ac2055f8007f829c1cd8c982eec45

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134680 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7135 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45957 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142204 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86618 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/302a7d9b-405a-49db-b2d2-95af54ccad24) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/136550 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7737 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6988 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102934 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/70213 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137627 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5434 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120741 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83739 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/82225cc4-3ad8-4712-9da4-8badad83d138) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5287 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2902 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/2794 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/126725 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114513 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38888 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144898 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/133188 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6809 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39469 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111330 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6883 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5712 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111625 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28337 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5123 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117021 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60693 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6860 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35205 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/166097 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6664 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70441 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43413 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6900 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6773 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Configurations/CommonBase.xcconfig:
```diff
@@ -65,6 +65,9 @@ _EXPERIMENTAL_SWIFT_EXPLICIT_MODULES = $(WK_DEFAULT_SWIFT_ENABLE_EXPLICIT_MODULE
 WK_DEFAULT_SWIFT_ENABLE_EXPLICIT_MODULES = YES;
 WK_DEFAULT_SWIFT_ENABLE_EXPLICIT_MODULES[sdk=macosx14*] = ;
 
+WK_DEFAULT_COMPILATION_CACHE_ENABLE_CACHING = YES;
```
I feel like we should limit it to Xcode 26 or newer.

emw-apple replied:
It's a new build setting as of Xcode 26, added in April 2025: <https://github.com/swiftlang/swift-build/commit/f974e0dc3841d82df08ead210fb900f850b919ac>.

emw-apple committed:
[Xcode] Enable compilation caching by default
https://bugs.webkit.org/show_bug.cgi?id=303579
rdar://131590077

Reviewed by Alexey Proskuryakov.

Toggle COMPILATION_CACHE_ENABLE_CACHING in CommonBase.xcconfig. Add a
script phase that runs in WTF which ensures that WebKit's opt-in ccache
support is not simultaneously enabled -- the two caches are redundant
when used together.

* Configurations/CommonBase.xcconfig:
* Makefile.shared: Remove logic that would automatically set
  WK_USE_CCACHE.
* Source/WTF/Scripts/check-ccache-compilation-caching.sh: Added.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/mac/CocoaAccessibilityConstantsPrivate.h: Added.
* Tools/ccache/README.md: Leave a note about LLVM compilation caching
  being the new default.

Canonical link: https://commits.webkit.org/304172@main
